packages/js/data/src/products/selectors.ts

```diff
@@ -122,38 +121,24 @@ export const isPending = (
 	return false;
 };
 
-export const getPermalinkParts = createRegistrySelector(
-	( select ) => ( state: ProductState, productId: number ) => {
-		const product = select( 'core' ).getEntityRecord(
-			'postType',
-			'product',
-			productId,
-			// @ts-expect-error query object is not part of the @wordpress/core-data types yet.
-			{
-				_fields: [
-					'id',
-					'permalink_template',
-					'slug',
-					'generated_slug',
-				],
-			}
-		);
-		if ( product && product.permalink_template ) {
-			const postName = product.slug || product.generated_slug;
+export const getPermalinkParts = ( state: ProductState, productId: number ) => {
+	const product = state.data[ productId ];
```

Comment:
Nice!  This seems much better than calling a `select` here which won't push updates to subscribers.

I think one nice addition might be to add a resolver in `resolvers.js` as well since if `getProduct()` has not yet been called, this will always return null.  We could create pseudo race conditions as well even if we do have a `getProduct` call followed by a `getParmlinks` call.

```
// resolvers.js
export function* getPermalinkParts( productId: number ) {
	yield resolveSelect( STORE_NAME, 'getProduct', [ productId ] );
}

// Might also need this inside of `getProduct` to prevent double fetches
export function* getProduct( productId: number ) {
		...
		yield dispatch( STORE_NAME, 'finishResolution', 'getPermalinkParts', [
			productId,
		] );
		...
```

Reply:
Thanks for the feedback. The data store stuff is still new to me. Added in https://github.com/woocommerce/woocommerce/pull/36706/commits/af21617678e2969a1331e6a88b65559177855a9e